code/datums/components/crafting/guncrafting.dm

```diff
@@ -1,22 +1,165 @@
 	custom_materials = list(/datum/material/wood = MINERAL_MATERIAL_AMOUNT * 6)
-	icon = 'icons/obj/improvised.dmi'
 	icon_state = "riflestock"
 
-/obj/item/weaponcrafting/silkstring
+// ATTACHMENTS //
+
+/obj/item/weaponcrafting/attachment
+	//The following are buffs given to the weapon this is attached to
+	force = 0
+	sharpness = 0
+	bleed_force = 0
+	//if this changes the fire sound of the weapon, put it here
+	var/fire_sound = null
+	//If we can attach this to a bow
+	var/bow_suitable = TRUE
+	//If we can attach this to a standard gun, currently a useless var but will be important later
+	var/gun_suitable = TRUE
+	var/spread = 0
+	var/damage_multiplier = 0
+	var/speed_multiplier = 0
+	var/added_description = "<span class='info'>The item attached to this does absolutely nothing.</span>"
+
+// PRIMARY //
+// Primary attachments are meant to affect the mechanical aspect of a weapon, for now, since im working on bows, im using this to separate bow string from other attachments
+// knowing this can be usefull for the future when its expanded into other weapons
+
+/obj/item/weaponcrafting/attachment/primary/cablestring
+	name = "cable coil string"
+	desc = "A cable coil string to be used in weapon crafting."
+	icon_state = "cablestring"
+	damage_multiplier = 0.5
+	speed_multiplier = 2
+	gun_suitable = FALSE
+	//Half damage and Half speed
+	added_description = "<span class='info'>The bowstring is improvised out of cable. It looks rather weak.</span>"
+
+/obj/item/weaponcrafting/attachment/primary/bamboostring
+	name = "bamboo fiber string"
+	desc = "A string made out of bamboo fiber to be used in weapon crafting."
+	icon_state = "bamboostring"
+	damage_multiplier = 0.8
+	speed_multiplier = 0.8
+	gun_suitable = FALSE
+	//A bit weak and a bit fast
+	added_description = "<span class='info'>The bowstring is made of bamboo fiber. Close to standard strength.</span>"
+
+/obj/item/weaponcrafting/attachment/primary/silkstring
 	name = "silkstring"
-	desc = "A long piece of Silk that looks like a cable coil."
-	icon = 'icons/obj/improvised.dmi'
+	desc = "A string made out of silk to be used in weapon crafting."
 	icon_state = "silkstring"
+	damage_multiplier = 1
+	speed_multiplier = 1
+	gun_suitable = FALSE
+	//Standard stuff, no debuffs
+	added_description = "<span class='info'>The bowstring is made of silkstring. Standard strength.</span>"
+
+/obj/item/weaponcrafting/attachment/primary/leatherstring
+	name = "leather string"
+	desc = "A string made out of leather to be used in weapon crafting."
+	icon_state = "leatherstring"
+	damage_multiplier = 1
+	speed_multiplier = 1.2
+	gun_suitable = FALSE
+	//Slight debuff to speed
+	added_description = "<span class='info'>The bowstring is made of leather. Standard strength.</span>"
+
+/obj/item/weaponcrafting/attachment/primary/sinewstring
+	name = "sinew string"
+	desc = "A bow string made out of sinew to be used in weapon crafting."
+	icon_state = "sinewstring"
+	damage_multiplier = 1.2
+	speed_multiplier = 1.2
```

Comment:
```suggestion
	damage_multiplier = 1.1
	speed_multiplier = 0.8
```

I think sinew should just be the best option outright, due to being the actual best material from this selection and also being mechanically the hardest to obtain for general usage.

Reply:
I had the design of lavaland materials being heavier, sturdier, so more damage, but less speed or other debuffs to compensate. Do you not like this approach? I'd like them all to be unique

Reply:
I'm on the fence about suggesting that the speed/damage multiplier be merged into a single value because faster should mean stronger from an immersion and sensibility standpoint, and likewise slower should mean weaker generally.

Slight variations one way or the other would be fine, but I do firmly believe that sinew, being the only especially good bowstring material among these, be the best.

Reply:
I actually honestly think speed and damage don't have to go hand in hand. Something thats flying fast will of course hurt more than if it was flying slower, but heavy objects wont usually be as fast as lighter objects. Which means they wont fly as fast yet hurt way more.

Its not unusual to see in gaming a large slow projectile that hurts most when you hit it. Bullet hell style. Being that SS13 is a technological space age sort of game I think seeing something like that would actually be pretty cool.

For me, having these values separate allows us to have more freedom in how we code weapons!

Also, regarding the sinew itself. I think in the event Security goes down to face Ashwalkers. A rain of slower projectiles would feel more... intense and cinematic than more instantaneous stuff.
I'd like to give people the chance to dodge the more dangerous projectiles in this case.

Reply:
Done!